src/Jobs/processUserData.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace BristolSU\UnionCloud\Jobs;
+
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use BristolSU\Support\User\User;
+
+class processUserData implements ShouldQueue
+{
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    protected array $user;
+    protected $repository;
+
+    /**
+     * Create a new job instance.
+     *
+     * @param array $user
+     */
+    public function __construct($user)
+    {
+        $this->user = $user;
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        $this->repository = app(\BristolSU\ControlDB\Contracts\Repositories\User::class);
+
+        $User = $this->repository->getById($this->user['uid']);
```

Comment:
This won't work because you're using the UnionCloud `uid` (the data user ID in portal terms) to find a control user by the control id. There's a `getByDataProviderId` function (or similar) which will get a control user by their data provider user (i.e. the unioncloud uid)

Reply:
That repository is the right one, as it's the control user repository and you want to get a control user. You're using `$this->repository->getById($controlId)`, but passing in the unioncloud (the data provider) id. You want to get a control user by the data provider ID, for which there's a `$this->repository->getByDataProviderId($this->user['uid'])` function.

This will return a control user still (if they exist), it's just using different criteria. This is needed because the control ID and data provider ID don't have to be the same.